[IR-2826] Unit Test - Spatial Input

heysokam opened the pull request:
## Summary

## Subtasks Checklist
- [x] InputPointerComponent
- [x] InputSinkComponent
- [x] InputSourceComponent
- [x] InputComponent
- [x] ClientInputHeuristics
- [ ] ClientInputFunctions
- [ ] ClientInputHooks
- [ ] ClientInputSystem

## Breaking Changes

## References
**Jira Trackers**:
https://tsu.atlassian.net/browse/IR-2826
https://tsu.atlassian.net/browse/IR-3281

## QA Steps

heysokam committed:
tst: UnitTests for `InputPointerComponent`

heysokam committed:
tst: UnitTests for `InputSinkComponent`

heysokam committed:
tst: UnitTests for `InputSourceComponent` (phase 0)

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
tst: UnitTests for `InputSourceComponent` (phase 1)

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
tst: UnitTests for `InputSourceComponent` (phase 2)

heysokam committed:
fix: Assert on `InputComponent.onSet` with `isArray` for `inputSinks`

heysokam committed:
tst: UnitTests for `InputComponent` (phase 0)

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
tst: Fix InputPointerComponent: `getPointerForCanvas` & `canvasEntity`

heysokam committed:
tst: Fix `InputSourceComponent.source.mapping` default value

heysokam committed:
tst: UnitTests for `InputPointerState`

heysokam committed:
fix: `InputPointerComponent.onRemove` stringified the hash incorrectly

heysokam committed:
new: `InputPointerState.createCameraPointerHash` & `CameraPointerHash`

heysokam committed:
tst: UnitTests for `InputPointerComponent.getPointerByID`

SamMazerIR commented:
looks good so far

SamMazerIR commented:
updated with dev, looks like some errors now:
`src/input/components/InputSourceComponent.test.tsx(199,27): error TS2314: Generic type 'ButtonStateMap' requires 1 type argument(s).
src/input/components/InputSourceComponent.test.tsx(294,33): error TS2314: Generic type 'ButtonStateMap' requires 1 type argument(s).
src/input/components/InputSourceComponent.test.tsx(324,33): error TS2314: Generic type 'ButtonStateMap' requires 1 type argument(s).
src/input/components/InputSourceComponent.test.tsx(389,33): error TS2314: Generic type 'ButtonStateMap' requires 1 type argument(s).
src/input/components/InputSourceComponent.test.tsx(419,33): error TS2314: Generic type 'ButtonStateMap' requires 1 type argument(s).
src/input/components/InputSourceComponent.test.tsx(450,33): error TS2314: Generic type 'ButtonStateMap' requires 1 type argument(s).`

heysokam commented:
@SamMazerIR
> updated with dev, looks like some errors now:

The errors were there before the merge 🙂. The branch is still half finished and I didn't get to apply the changes for that generic type that is erroring

I appreciate the early review, but accepting it was premature. The heavy weights of this PR (`InputComponent`, and `ClientInputSystem`) have not been started yet 👌🏻

heysokam committed:
tst: Fix for `ButtonStateMapAlias` generic type

heysokam committed:
tst: UnitTests for `InputComponent`: `toJson` and `@todo` list

heysokam committed:
fix: `InputSinkComponent.inputEntities` type being set to `never[]`

heysokam committed:
tst: UnitTests for `InputComponent.getInputEntities`

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
tst: UnitTests for `InputComponent.getInputSourceEntities`

heysokam committed:
tst: UnitTests: `InputExecutionOrder` and `InputExecutionSystemGroup`

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
doc: `@description` : `InputComponent.getMergedButtonsForInputSources`

heysokam committed:
tst: UnitTests for `InputComponent.getMergedButtonsForInputSources`

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
tst: UnitTests for `InputComponent.getMergedButtons`

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
tst: UnitTests for `InputComponent.getMergedAxesForInputSources`

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
tst: UnitTests for `InputComponent.getMergedAxes`

heysokam committed:
tst: Sketch for `InputComponent.useHasFocus` unit tests

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
tst: Fix UnitTests for `InputComponent.useHasFocus`

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
tst: Sketch for `InputComponent.useExecuteWithInput` test auto-gen

This function covers a matrix of 4*4*3 total branches
This commit is an early sketch for programmatic generation of each individual case

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
fmt: Expose more functions from the `ClientInputSystem.PRIVATE` object

heysokam committed:
tst: UnitTests for `ClientInputSystem.ts` (phase 0)

heysokam committed:
tst: UnitTests for `ClientInputSystem.PRIVATE.preventDefaultKeyDown`

heysokam committed:
fmt: Clause Guard for `ClientInputSystem.setInputSources` body

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
fix: Incorrect fail in `useExecuteWithInput` when `captureEntity == 0`

heysokam committed:
tst: Fix UnitTests for `InputComponent.useExecuteWithInput`

heysokam committed:
tst: UnitTests for `ClientInputSystem.PRIVATE.setInputSources`

heysokam committed:
fmt: Clause Guard and DRY for the `applyHeuristicProximity` function

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
doc: Internal `@note` comment about the DRY change from last commit

heysokam committed:
fmt: Some more Clause Guards around `ClientInputSystem.tsx`

heysokam committed:
fmt: Fix for new functions using global state without any need for it

heysokam committed:
tst: Internal notes for todo tests for ClientInputSystem

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
fmt: Change `bboxHitTarget` usage @`applyHeuristicBBoxes` to argument

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
fmt: Change `inputRay` usage by `applyHeuristicBBoxes` to an argument

heysokam committed:
doc: Remove temp comment from applyHeuristicBBoxes

heysokam committed:
fmt: Change `raycaster` usage by `applyHeuristicMeshes` to an argument

heysokam committed:
fmt: Change `raycaster` usage by `applyHeuristicEditor` to an argument

heysokam committed:
fmt: Remove global `quat` access from `applyRaycastedInputHeuristics`

heysokam committed:
fmt: Clarify global access objects with `g_` and remove nested access

heysokam committed:
tst: Internal wip todo list for `ClientInputSystem.test.tsx`

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

HexaField commented on packages/spatial/src/input/systems/ClientInputSystem.tsx:
```diff
@@ -85,7 +85,7 @@ const DRAGGING_THRESHOLD = 0.001
 
 /** anti-garbage variable!! value not to be used unless you set values just before use*/
-const pointerPositionVector3 = new Vector3()
+const g_pointerPositionVector3 = new Vector3()
```
we do not use this format for variables. please keep the naming conventions the same.

heysokam replied:
kk will change them.
What could I do to clarify that these variables are accessed from the top of the file, and not sent into the functions or created by them?

heysokam replied:
Fixed with 25c6b83

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
fmt: Refactor `ClientInputSystem.tsx` into multiple modules

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
tst: Split `ClientInputSystem.test.tsx` into multiple modules

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
tst: Update the `@todo` list of ClientInput tests

heysokam committed:
fmt: Move `useXRInputSources` up, next to the `use*` declarations

heysokam committed:
tst: UnitTests for `ClientInputFunctions.cleanupButton` & improve todo

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
tst: Reference notes for `applyProximity` unit tests todo/impl

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
tst: UnitTests for `ClientInputHeuristics.applyRaycastedInput`

heysokam committed:
fmt: Refactor `ClientInputSystem.tsx` into multiple files

heysokam committed:
fix: Canvas being possibly null error

heysokam committed:
fix: Canvas being possibly null error

heysokam committed:
tst: Improve todo list comments for ClientInputHeuristics tests

heysokam committed:
tst: Update `MockEventListener` to include `type` and `hasEvent` check

heysokam committed:
fix: Missing `document.removeEventListener` @useNonSpatialInputSources

heysokam committed:
tst: Bugfix for `MockEventListener.hasEvent` on empty arrays

heysokam committed:
tst: UnitTests for `ClientInputHooks.useNonSpatialInputSources`

heysokam committed:
tst: UnitTests for `ClientInputHooks.useGamepadInputSources`

heysokam committed:
tst: Sketch for `ClientInputHooks.useXRInputSources` unit tests

heysokam committed:
Merge branch 'dev' into IR-3297-client-input-system-restructure-into-multiple-files

heysokam committed:
tst: Fix UnitTests for `ClientInputHooks.useXRInputSources`

heysokam committed:
tst: Update todo lists for the ClientInput tests

heysokam committed:
tst: Sketch for `ClientInputHooks.CanvasInputReactor` mock data setup

heysokam committed:
tst: UnitTests for `ClientInputHeuristics.applyBBoxes`

heysokam committed:
tst: UnitTests for `ClientInputHooks.CanvasInputReactor`

heysokam committed:
tst: UnitTests for `ClientInputHeuristics.applyPhysicsColliders`

heysokam committed:
tst: Sketch for `ClientInputHeuristics.applyMeshes` tests

heysokam committed:
tst: UnitTests for `ClientInputHooks.MeshInputReactor` (phase 0)

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
tst: UnitTests for `ClientInputHooks.BoundingBoxInputReactor` (phase 0

heysokam committed:
tst: Fix UnitTests for `ClientInputFunctions.preventDefaultKeyDown`

heysokam committed:
tst: UnitTests `ClientInputHeuristics.applyMeshes`

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input

heysokam committed:
tst: UnitTests `ClientInputHeuristics.applyEditor`

heysokam committed:
tst: Improve internal todo for `applyXRUI` and `applyProximity`

heysokam committed:
tst: Utilities to mock XRUI `WebContainer3D` inside NodeJS/Mocha

heysokam committed:
tst: UnitTests for `ClientInputHeuristics.applyEditor`

heysokam committed:
fmt: Rename `apply*` to `find*`. `Heuristic`'s etymology is `find out`

The functions are not applying anything.
They are finding entities, and adding them into a dataset, along with their distance to the relevant input source.

Heuristic Etimology:
from Greek `heuriskein` "to find; find out, discover;"
`-ist` suffix: "one who does or makes"
`-ic` suffix: "having to do with"
The above combined: "The thing that has to do with finding out / discovery"
https://www.etymonline.com/word/heuristic

heysokam committed:
doc: Reword internal comment on `findProximity` if entity is undefined

heysokam committed:
tst: Finalize the todo list for `ClientInputHeuristics.findProximity`

heysokam committed:
tst: Sketch for `ClientInputHeuristics.findProximity` tests

heysokam committed:
tst: Add root `nyc` config file for test-coverage options

heysokam committed:
tst: Fix `nyc` incorrectly instrumenting `.test.ts*` files

heysokam committed:
tst: UnitTests for `InputPointerComponent.usePointersForCamera`

heysokam committed:
fmt: Create temporary bkp file for `ClientInputSystem.tsx`

heysokam committed:
git: Revert refactoring changes

heysokam committed:
Merge branch 'dev' into IR-3297-client-input-system-restructure-into-multiple-files

heysokam committed:
fmt: Remove `ClientInputSystem.bkp.tsx` temporary file

heysokam committed:
fmt: Re-refactor `ClientInputSystem` into multi-file from latest dev

heysokam committed:
Merge branch 'IR-3297-client-input-system-restructure-into-multiple-files' into IR-2828-unit-test-spatial-input

heysokam committed:
tst: Small fixes around the SpatialInput tests

heysokam committed:
tst: UnitTests for `ClientInputHeuristics.findProximity`

heysokam committed:
tst: UnitTests for `ClientInputFunctions.assignInputSources` (phase 1)

heysokam committed:
Merge branch 'dev' into IR-2828-unit-test-spatial-input
